pkgs/development/python-modules/pytorch/default.nix

```diff
@@ -30,23 +30,58 @@
   isPy3k, pythonOlder }:
 
 let
-  inherit (cudaPackages) cudatoolkit cudnn nccl;
+  inherit (cudaPackages) cudnn nccl;
 in
 
 # assert that everything needed for cuda is present and that the correct cuda versions are used
-assert !cudaSupport || (let majorIs = lib.versions.major cudatoolkit.version;
+assert cudaSupport -> (let majorIs = cudaPackages.cudaMajorVersion;
                         in majorIs == "9" || majorIs == "10" || majorIs == "11");
 
 # confirm that cudatoolkits are sync'd across dependencies
-assert !(MPISupport && cudaSupport) || mpi.cudatoolkit == cudatoolkit;
-assert !cudaSupport || magma.cudatoolkit == cudatoolkit;
+# TODO: verify it's OK to equality-compare cudaPackages (which are attrsets, scopes, or whatever)
```

Comment:
Where do we stand on this question? Has anyone been able to run a quick test?

Reply:
With help from [#nix-lang:nixos.org](https://matrix.to/#/#nix-lang:nixos.org):

- `==` evaluates terms into values
- ...and proceeds with [eqValues](https://github.com/NixOS/nix/blob/c9a29d0d92529338026190374d0e7e886f486e47/src/libexpr/eval.cc#L1602-L1607)

The latter:

- will probably immediately return `true` in absence of overrides, because of `&v1 == &v2` special case
- otherwise recurse intro the attrset
  - for attributes-derivations this would cause evaluation of their `outPath`s
  - for functions (`overrideScope'`?) it would immediately return `false`

TLDR: I think it means that we'll essentially require pointer-equality. I can be wrong